.github/workflows/test.yml

```diff
@@ -208,6 +240,7 @@ jobs:
     name: Regenerate stateful disks
     runs-on: ubuntu-latest
     needs: build
+    if: ${{ needs.build.outputs.any_changed == 'true' || github.event.inputs.regenerate-disks == 'true'}}
```

Comment:
Question: do we also want to regenerate disks when the workflow file changes?

Reply:
This would add a few hours to most PRs, if we'd like to do this for specific files it would be best to add those to this step: https://github.com/ZcashFoundation/zebra/pull/4074/files#diff-faff1af3d8ff408964a57b2e475f69a6b7c7b71c9978cccc8f471798caac2c88R77

Reply:
It seems like the long-term solution is to split this job and its dependencies into their own workflow file.

Do we have a ticket for that?

Reply:
I'll create the ticket for `reusable workflows` which, by the way, is a considerable change to be done before working in pipeline documentation. I think it's even better to prioritize that somehow.